core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsRaftManager.java

```diff
@@ -88,6 +95,29 @@ public String raftId() {
       return raftId;
    }
 
+   @Override
+   public CompletionStage<Void> addMember(String raftId) {
+      AggregateCompletionStage<Void> stage = CompletionStages.aggregateCompletionStage();
+      raftStateMachineMap.values().forEach(raftChannel -> stage.dependsOn(raftChannel.addMember(raftId)));
+      return stage.freeze().whenComplete((ignore, t) -> {
+         if (t == null) raftMembers.add(raftId);
+      });
+   }
+
+   @Override
+   public CompletionStage<Void> removeMembers(String raftId) {
+      AggregateCompletionStage<Void> stage = CompletionStages.aggregateCompletionStage();
+      raftStateMachineMap.values().forEach(raftChannel -> stage.dependsOn(raftChannel.removeMember(raftId)));
+      return stage.freeze().whenComplete((ignore, t) -> {
+         if (t == null) raftMembers.remove(raftId);
+      });
+   }
+
+   @Override
+   public Collection<String> raftMembers() {
+      return Collections.unmodifiableCollection(raftMembers);
```

Comment:
probably worth to merge with the members in the RAFT protocol
```suggestion
      var stateMembers = raftStateMachineMap.values().stream().findFirst()
            .map(jgroupsRaftChannel -> jgroupsRaftChannel.raftHandle.raft().members())
            .orElse(Collections.emptyList());
      return Stream.concat(stateMembers.stream(), raftMembers.stream()).collect(Collectors.toSet());
```